optimum/exporters/tasks.py

```diff
@@ -1125,42 +1151,58 @@ def get_supported_tasks_for_model_type(
                 The name of the exporter.
             model_name (`Optional[str]`, defaults to `None`):
                 The name attribute of the model object, only used for the exception message.
-            library_name (defaults to `transformers`):
-                 The library name of the model.
+            library_name (`Optional[str]`, defaults to `None`):
+                The library name of the model. Can be any of "transformers", "timm", "diffusers", "sentence_transformers".
 
         Returns:
             `TaskNameToExportConfigDict`: The dictionary mapping each task to a corresponding `ExportConfig`
             constructor.
         """
+        if library_name is None:
+            logger.warning(
+                'Not passing the argument `library_name` to `get_supported_tasks_for_model_type` is deprecated and the support will be removed in a future version of Optimum. Please specify a `library_name`. Defaulting to `"transformers`.'
```

Comment:
I feel the warning a little bit misleading, feeling like that we are about to deprecate the func which is not the case. Maybe just rephrase something some "`library_name` will be mandatory"?

Reply:
Thank you, fixed in https://github.com/huggingface/optimum/pull/1647/commits/665c880d547a3664fe60fd8ca8d931d5b68b1b73